ledger/stateproofverificationtracker.go

```diff
@@ -0,0 +1,244 @@
+	}
+
+	spt.stateProofVerificationMu.Lock()
+	defer spt.stateProofVerificationMu.Unlock()
+
+	if blk.Round()%currentStateProofInterval == 0 {
+		spt.insertCommitData(&blk)
+	}
+
+	if delta.StateProofNext != 0 {
+		spt.insertDeleteData(&blk, &delta)
+	}
+}
+
+func (spt *stateProofVerificationTracker) committedUpTo(round basics.Round) (minRound, lookback basics.Round) {
+	return round, 0
+}
+
+func (spt *stateProofVerificationTracker) produceCommittingTask(_ basics.Round, _ basics.Round, dcr *deferredCommitRange) *deferredCommitRange {
+	return dcr
+}
+
+func (spt *stateProofVerificationTracker) prepareCommit(dcc *deferredCommitContext) error {
+	spt.stateProofVerificationMu.RLock()
+	defer spt.stateProofVerificationMu.RUnlock()
+
+	lastDataToCommitIndex := spt.committedRoundToLatestCommitDataIndex(dcc.newBase)
+	dcc.stateProofVerificationCommitData = make([]verificationCommitData, lastDataToCommitIndex+1)
+	copy(dcc.stateProofVerificationCommitData, spt.trackedCommitData[:lastDataToCommitIndex+1])
+
+	dcc.stateProofVerificationLatestDeleteDataIndex = spt.committedRoundToLatestDeleteDataIndex(dcc.newBase)
+	if dcc.stateProofVerificationLatestDeleteDataIndex > -1 {
+		dcc.stateProofVerificationDeleteData = spt.trackedDeleteData[dcc.stateProofVerificationLatestDeleteDataIndex]
+	}
+
+	return nil
+}
+
+func (spt *stateProofVerificationTracker) commitRound(ctx context.Context, tx *sql.Tx, dcc *deferredCommitContext) (err error) {
+	err = insertStateProofVerificationData(ctx, tx, &dcc.stateProofVerificationCommitData)
+	if err != nil {
+		return err
+	}
+
+	if dcc.stateProofVerificationLatestDeleteDataIndex > -1 {
+		err = deleteOldStateProofVerificationData(ctx, tx, dcc.stateProofVerificationDeleteData.stateProofNextRound)
+	}
+
+	return err
+
+}
+
+func (spt *stateProofVerificationTracker) postCommit(_ context.Context, dcc *deferredCommitContext) {
+	spt.stateProofVerificationMu.Lock()
+	defer spt.stateProofVerificationMu.Unlock()
+
+	spt.trackedCommitData = spt.trackedCommitData[len(dcc.stateProofVerificationCommitData):]
+	spt.trackedDeleteData = spt.trackedDeleteData[dcc.stateProofVerificationLatestDeleteDataIndex+1:]
+}
+
+func (spt *stateProofVerificationTracker) postCommitUnlocked(context.Context, *deferredCommitContext) {
+
+}
+
+func (spt *stateProofVerificationTracker) handleUnorderedCommit(*deferredCommitContext) {
+
+}
+
+func (spt *stateProofVerificationTracker) close() {
+	if spt.dbQueries.lookupStateProofVerificationData != nil {
+		spt.dbQueries.lookupStateProofVerificationData.Close()
+	}
+}
+
+func (spt *stateProofVerificationTracker) LookupVerificationData(stateProofLastAttestedRound basics.Round) (*ledgercore.StateProofVerificationData, error) {
+	spt.stateProofVerificationMu.RLock()
+	defer spt.stateProofVerificationMu.RUnlock()
+
+	if len(spt.trackedCommitData) == 0 || stateProofLastAttestedRound < spt.trackedCommitData[0].verificationData.TargetStateProofRound {
+		return spt.dbQueries.lookupData(stateProofLastAttestedRound)
```

Comment:
lookupData might return `sql.ErrNoRows` so I think it would be nice to return `errStateProofVerificationDataNotFound` instead